cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -675,6 +674,28 @@ class FieldInstruction extends Instruction {
   }
 }
 
+/**
+ * An `Instruction` that is associated with a particular `Parameter`.
+ */
+class ParameterInstruction extends Instruction {
+  Language::Parameter parameter;
+
+  ParameterInstruction() {
+    parameter = Construction::getInstructionParameter(this)
+  }
+
+  override string getImmediateString() {
+    result = parameter.toString()
+  }
+
+  /**
+   * Gets the `Parameter` associated with this `Instruction`
+   */
+  final Language::Parameter getParameter() {
```

Comment:
After your last update, it looks like all uses of `getParameter` and `getInstructionParameter` are gone. All that's left is the code for passing them between various stages of the IR.

Reply:
After the refactor to make the caller side effect nodes argument-based, `InitializeParameterInstruction` is the only class extending `ParameterInstruction`. I could remove `ParameterInstruction` and revert `InitializeParameterInstruction`, but I'm expecting that the callee side of this logic will add more classes that extend `ParameterInstruction`.

Reply:
There's already `VariableInstruction` for associating an instruction with a variable, and `InitializeParameterInstruction` was using that before. Shouldn't that suffice for the callee-side side effect instructions too?